File: 908-middle-of-the-linked-list/middle-of-the-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head) {
    
    struct ListNode *slow=head,*fast=head;

    while(fast!=NULL && fast->next!=NULL){

        slow=slow->next;
        fast=fast->next;
        fast=fast->next;

    } 
    return slow;
  
}


/*
  struct ListNode* crr=head;
    int count=0;

    while(crr!=NULL){
        count++;
        crr=crr->next;
    }
    int mid=count/2+1;

    crr=head;
    count=1;
    while(count<mid){
        count++;
        crr=crr->next;
    }
    return crr;
*/

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        
        ListNode crr=head;
        int len=0;
        while(crr!=null){
            len++;
            crr=crr.next;
        }
        
        int mid=len/2+1;

        crr=head;
        int count=1;
        while(count<mid){
            count++;
            crr=crr.next;
        }

        return crr;







    }
}